packages/ui/src/components/va-tabs/VaTabs.vue

```diff
@@ -60,350 +60,34 @@
-
-    this.$emit('click:next')
-  }
-
-  moveToTab (tab: VaTab) {
-    if (this.showPagination && tab.leftSidePosition + this.containerRef.clientWidth <= this.tabsRef.clientWidth) {
-      this.tabsContentOffset = tab.leftSidePosition
-    } else if (this.showPagination && tab.rightSidePosition >= this.containerRef.clientWidth) {
-      this.tabsContentOffset = tab.rightSidePosition - this.containerRef.clientWidth
-    } else {
-      this.tabsContentOffset = 0
-    }
-  }
-
-  updateSlider (tab: VaTab) {
-    if (this.$props.vertical) {
-      const sliderOffsetY = this.containerRef.clientHeight - tab.$el.offsetTop - tab.$el.clientHeight
-      this.sliderOffsetY = Math.max(sliderOffsetY, 0)
-      this.sliderHeight = tab.$el.clientHeight
-      this.sliderOffsetX = 0
-      this.sliderWidth = null
-    } else {
-      this.sliderOffsetX = tab.$el.offsetLeft
-      this.sliderWidth = tab.$el.clientWidth
-      this.sliderOffsetY = 0
-      this.sliderHeight = null
-    }
-  }
-
-  resetSliderSizes () {
-    this.sliderWidth = 0
-    this.sliderHeight = 0
-  }
-
-  updateStartingXPoint () {
-    this.startingXPoint = 0
-    if (!this.showPagination) {
-      return
-    }
-    if (this.$props.right) {
-      this.startingXPoint = this.tabsRef?.clientWidth - this.containerRef?.clientWidth
-    } else if (this.$props.center) {
-      this.startingXPoint = Math.floor((this.tabsRef?.clientWidth - this.containerRef?.clientWidth) / 2)
-    }
-  }
-
-  includeAnimation () {
-    if (!this.animationIncluded) {
-      requestAnimationFrame(() => {
-        this.animationIncluded = true
-      })
-    }
-  }
-
-  redrawTabs () {
-    const oldShowPaginationValue = this.showPagination
-    this.updatePagination()
-    if (oldShowPaginationValue === this.showPagination) {
-      this.updateTabsState()
-      this.includeAnimation()
-    } else {
-      requestAnimationFrame(() => {
-        this.updateTabsState()
-        this.includeAnimation()
-      })
-    }
-  }
-
-  mounted () {
-    this.redrawTabs()
-    requestAnimationFrame(() => {
-      this.resizeObserver = new ResizeObserver(this.redrawTabs)
-      this.resizeObserver.observe(this.wrapperRef)
-      this.resizeObserver.observe(this.tabsRef)
-    })
-  }
-
-  beforeUnmount () {
-    if (this.resizeObserver) {
-      this.resizeObserver.disconnect()
-    }
-  }
-}
+  props: {
+    ...useStatefulProps,
+    modelValue: { type: [String, Number], default: null },
+    left: { type: Boolean, default: true },
+    right: { type: Boolean, default: false },
+    center: { type: Boolean, default: false },
+    grow: { type: Boolean, default: false },
+    disabled: { type: Boolean, default: false },
+    hideSlider: { type: Boolean, default: false },
+    vertical: { type: Boolean, default: false },
+    color: { type: String, default: 'primary' },
+    prevIcon: { type: String, default: 'chevron_left' },
+    nextIcon: { type: String, default: 'chevron_right' },
+  },
+
+  setup: useTabsView,
```

Comment:
useTabsView hook can be setup function instead of hook. Then you will not be forced to write extra ts interfaces.

Reply:
Additionally, this moves all logic from component to separate file, making it harder to navigate.
Separating into file is generally meant to either cut features away from component when it's too large or for reuse.

Reply:
Restored all components logic back to itself.